### PR TITLE
Fixed offload_arch cflags with targetID HIP

### PR DESCRIPTION
**Pull request by zjing14**
- To fix the issue: SWDEV-259381 that introduces new offload_arch flags for all supported devices, which causes the following error
`fatal error: error in backend: Cannot select: intrinsic %llvm.amdgcn.mfma.f32.32x32x4f16`
- Remove all offload_arch flags in HIP_COMPILER_FLAGS, and then add offload_arch flag back with the target device.

### CODE REVIEW

File: src/hip/hip_build_utils.cpp [hunk 1]
```diff
@@ -120,6 +120,7 @@ boost::filesystem::path HipBuild(boost::optional<TmpDir>& tmp_dir,
 
     params += " -Wno-unused-command-line-argument -I. ";
     params += MIOPEN_STRINGIZE(HIP_COMPILER_FLAGS);
+    params += " --" + std::string(MIOPEN_ARCH_FLAG) + "=" + dev_name;
```
Comment (atamazov) [inline src/hip/hip_build_utils.cpp:123]: Move it to line 115 (i.e. replace line 115 with it). And what about line 116 ("--cuda-device-only")?

**Review by atamazov — CHANGES_REQUESTED** [review]

**Comment by atamazov** [comment]
@t-tye Please let me know if COMgr path also needs to be fixed.

**Comment by t-tye** [comment]
> 
> 
> @t-tye Please let me know if COMgr path also needs to be fixed. I guess not -- right now we are not passing anything like "---cuda..." to COMgr.

COMgr takes the name of the target as a target ID now. So you can askp for a "gfx908:sramecc+:xnack-". If you ask simply for "gfx908" you will get a code object that will work with any setting of sramecc or xnack. So does the code currently specify the xnack/sramecc settings? If so the old method still works (using -mxnack/-msramecc) but the new method can be used instead.

**Comment by zjing14** [comment]
CI passed.

**Review by pfultz2 — APPROVED** [review]

**Comment by daniellowell** [comment]
@zjing14 @atamazov is this needed still?

**Comment by atamazov** [comment]
@daniellowell I guess so.

**Comment by daniellowell** [comment]
@zjing14 is this still required?

**Comment by atamazov** [comment]
Oops, SWDEV-259381 is closed. I think I or someone else has already fixed this.

**Pull request closed without merging**